### PR TITLE
[DRAFT] Updates to workspace profile

### PR DESCRIPTION
**Pull request by binaek**
This PR updates:
* Adds properties to `workspace profile`
* Adds `query` options block to `workspace profile`

Pending:
* Deprecate global `terminal` options
* Add `check` options block to `workspace profile`
* Add test cases to verify behaviour

### CODE REVIEW

File: pkg/steampipeconfig/modconfig/workspace_profile.go [hunk 1]
```diff
@@ -19,8 +21,13 @@ type WorkspaceProfile struct {
...
 	Base              *WorkspaceProfile `hcl:"base"`
+	SearchPath        *string           `hcl:"search_path" cty:"search_path"`
+	SearchPathPrefix  *string           `hcl:"search_path_prefix" cty:"search_path_prefix"`
+	Watch             *bool             `hcl:"watch" cty:"watch"`
+	MaxParallel       *int              `hcl:"max_parallel" cty:"max-parallel"`
 
 	// options
+	QueryOptions      *options.Query
 	GeneralOptions    *options.General
 	TerminalOptions   *options.Terminal
```
Comment (kaidaguerre) [inline pkg/steampipeconfig/modconfig/workspace_profile.go:32]: delete terminal options

**Review by kaidaguerre — CHANGES_REQUESTED** [review]

File: pkg/cmdconfig/cmd_flags.go
Comment (binaek) [inline pkg/cmdconfig/cmd_flags.go:1]: unused arg opt

File: pkg/steampipeconfig/modconfig/workspace_profile.go [hunk 2]
```diff
@@ -125,22 +192,42 @@ func (p *WorkspaceProfile) ConfigMap() map[string]interface{} {
...
 
-	// now add options
-	// build flat config map with order or precedence (low to high): general, terminal, connection
-	// this means if (for example) 'search-path' is set in both terminal and connection options,
-	// the value from connection options will have precedence
-	// however, we also store all values scoped by their options type, so we will store:
-	// 'database.search-path', 'terminal.search-path' AND 'search-path' (which will be equal to 'terminal.search-path')
-	if p.GeneralOptions != nil {
-		res.PopulateConfigMapForOptions(p.GeneralOptions)
+	if cmd.Name() == "query" && p.QueryOptions != nil {
```
Comment (kaidaguerre) [inline pkg/steampipeconfig/modconfig/workspace_profile.go:206]: can we add consts for command names please

File: pkg/steampipeconfig/modconfig/workspace_profile.go [hunk 3]
```diff
@@ -125,22 +192,42 @@ func (p *WorkspaceProfile) ConfigMap() map[string]interface{} {
...
+		res.PopulateConfigMapForOptions(p.DashboardOptions)
 	}
 
 	return res
 }
+
+// stringToSlice checks that `str` is `nil` and returns a string slice with `str`
+// separated with `separator`
+// If `str` is `nil`, this returns a `nil`
+func stringToSlice(str *string, separator string) []string {
```
Comment (kaidaguerre) [inline pkg/steampipeconfig/modconfig/workspace_profile.go:222]: can we either move to utils and remove reference to search path, or rename to `searchPathFromString`

I think as we have the strip whitespace, that this is really a search path specific function

File: pkg/steampipeconfig/options/check.go [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+		if t.Separator == nil && o.Separator != nil {
+			t.Separator = o.Separator
+		}
+		if t.Header == nil && o.Header != nil {
+			t.Header = o.Header
+		}
+	}
+}
+
+// ConfigMap :: create a config map to pass to viper
```
Comment (kaidaguerre) [inline pkg/steampipeconfig/options/check.go:39]:
```
ConfigMap :: create a config map to pass to viper
```

We do not use this comment syntax any more - please update to 
```
ConfigMap creates a config map to pass to viper
```

File: pkg/steampipeconfig/options/dashboard.go [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+		return
+	}
+	if o, ok := otherOptions.(*WorkspaceProfileDashboard); ok {
+		if t.Browser == nil && o.Browser != nil {
+			t.Browser = o.Browser
+		}
+	}
+}
+
+// ConfigMap :: create a config map to pass to viper
```
Comment (kaidaguerre) [inline pkg/steampipeconfig/options/dashboard.go:33]: comment format

File: pkg/steampipeconfig/options/query.go [hunk 6]
```diff
@@ -0,0 +1,137 @@
...
+		if t.Timing == nil && o.Timing != nil {
+			t.Timing = o.Timing
+		}
+		if t.AutoComplete == nil && o.AutoComplete != nil {
+			t.AutoComplete = o.AutoComplete
+		}
+	}
+}
+
+// ConfigMap :: create a config map to pass to viper
```
Comment (kaidaguerre) [inline pkg/steampipeconfig/options/query.go:47]: comment format

**Review by kaidaguerre — CHANGES_REQUESTED** [review]

File: pkg/steampipeconfig/modconfig/workspace_profile.go (hunk 2 above)
Comment (binaek) [reply pkg/steampipeconfig/modconfig/workspace_profile.go:206]: :+1:

**Pull request merged**